packages/nextjs/src/utils/instrumentServer.ts

```diff
@@ -329,14 +329,21 @@ function makeWrappedMethodForGettingParameterizedPath(
   origMethod: ApiPageEnsurer | PageComponentFinder,
 ): WrappedApiPageEnsurer | WrappedPageComponentFinder {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const wrappedMethod = async function (this: Server, parameterizedPath: string, ...args: any[]): Promise<any> {
+  const wrappedMethod = async function (
+    this: Server,
+    parameterizedPath: string | { pathname: string },
```

Comment:
I think we should add a comment here explaining that the two different types come from two different nextjs versions.